lib/types/any/index.js

```diff
@@ -12,8 +12,9 @@ let Cast = null;
 // Declare internals
 
 const internals = {
-    Set: require('../../set')
+    Set,
```

Comment:
You lose a lot of functionality trying to use a native `Set` instead of the `set.js` implementation, especially around Refs.